tests/unit/test_pytest_operator.py

```diff
@@ -291,3 +291,34 @@ async def mock_run(*cmd):
     ops_test = plugin.OpsTest(Mock(**{"module.__name__": "test"}), tmp_path_factory)
     await ops_test.create_crash_dump()
     mock_log.info.assert_any_call("juju-crashdump command was not found.")
+
+
+def test_no_deploy_mode(testdir):
+    """Test running no deploy mode."""
+    testdir.makepyfile(
+        """
+        import pytest
+
+        @pytest.mark.abort_on_fail
+        @pytest.mark.skip_if_deployed
+        def test_build_and_deploy():
+            pass
+
+        def test_01():
+            pass
+
+        def test_02():
+            pass
+    """
+    )
+    # test without --no-deploy option
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=3)
+
+    # test with --no-deploy, but without --model option
+    result = testdir.runpytest("--no-deploy")
+    result.assert_outcomes(passed=3)
+
+    # test with --no-deploy and --model
+    result = testdir.runpytest("--no-deploy", "--model", "test-model")
```

Comment:
related to comment above, can add a test here checking behaviour when we do `--no-deploy` without `--model`